designs/node-repair.md

```diff
@@ -0,0 +1,92 @@
+#  Karpenter Node Auto Repair 
+
+## Problem Statement
+
+Karpenter users are observing that occasionally nodes can face failure mode that prevent them to be healthy in a Kubernetes clusters. These nodes stay in unhealthy state for a period of time without any actions, and in some cases forever with manual intervention. Karpenter continues to simulate pods to these nodes, but the Kubernetes scheduler won't schedule to them. This will leave pods unscheduled and cause downtime for applications. Customers have the need for karpenter to repair unhealthy nodes. The karpenter team has seen several failure mode such as disk pressure, memory pressure, networking outage etc. that can cause nodes to get into an unhealthy state Here are issues related to the topic:
```

Comment:
I'd be cautious of repairing on disk/memory pressure, as opposed to letting taint based eviction do their job https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-based-evictions. That said, I do strongly agree there are cases where the only fix is to toss the node.